TrackletAlgorithm/MatchCalculator.h

```diff
@@ -788,27 +1126,60 @@ void MatchCalculator(BXType bx,
     ap_int<13> proj_z_corr   = proj_z + z_corr;      // original proj z plus z correction
 
     // Get phi and z difference between the projection and stub
-    ap_int<10> delta_z         = stub_z - proj_z_corr;
+    ap_int<12> delta_z        = stub_z - proj_z_corr;
     ap_int<14> delta_z_fact   = delta_z * kFact;
     const ap_int<18> &stub_phi_long  = stub_phi;         // make longer to allow for shifting
     const ap_int<18> &proj_phi_long  = proj_phi_corr;    // make longer to allow for shifting
     ap_int<18> shiftstubphi   = stub_phi_long << kPhi0_shift;                        // shift
+    if(isDisk && isPSStub)
+      shiftstubphi = stub_ps_phi << kPhi0_shift;
+    else if(isDisk && !isPSStub) {
+      shiftstubphi = stub_2s_phi << kPhi0_shift;
+    }
     ap_int<18> shiftprojphi   = proj_phi_long << (kShift_phi0bit - 1 + kPhi0_shift); // shift
-    ap_int<17> delta_phi      = shiftstubphi - shiftprojphi;
-    ap_uint<17> abs_delta_phi = iabs<17>( delta_phi );    // absolute value of delta phi
+    constexpr int dphibit = 20;
+    ap_int<dphibit> delta_phi      = shiftstubphi - shiftprojphi;
+    ap_uint<3> shiftprojz     = 7;
+    ap_int<7> proj_r_corr    = (stub_z * proj_zd) >> shiftprojz;
+    if(isDisk && isPSStub)
+      proj_r_corr = (stub_ps_z * proj_zd) >> shiftprojz;
+    else if(isDisk)
+      proj_r_corr = (stub_2s_z * proj_zd) >> shiftprojz;
+    const ap_uint<13> &proj_r_long  = proj_z + proj_r_corr;
+    ap_uint<1> shiftr         = 1;
+    ap_int<12> delta_r        = (stub_r >> shiftr) - proj_r_long; // proj_z = RZ
+    ap_uint<12> tmp_stubr = isProjDisk ? LUT_matchcut_rDSS[ap_uint<12>(stub_2s_r)] : LUT_matchcut_rDSS[ap_uint<12>(stub_r)]; //FIXME
```

Comment:
Same comments as for MP : LUT_matchcut_rDSS is only 16 deep.

Reply:
Fixed